test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MockServer.cs

```diff
@@ -0,0 +1,440 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Security.Principal;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using NuGet.Common;
+using NuGet.Protocol;
+using NuGet.Test.Server;
+using Test.Utility;
+
+namespace Dotnet.Integration.Test
+{
+    /// <summary>
+    /// A Mock Server that is used to mimic a NuGet Server.
+    /// </summary>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1001:TypesThatOwnDisposableFieldsShouldBeDisposable")]
+    public class MockServer : IDisposable
```

Comment:
I tried to address this comment, while the effort is relatively bigger than I expect. Since it only affects the testing, can I do it gradually,  in a separate PR?

I tried in branch dev-hengliu-xplat3-feature-dotnet-nuget-sign-test2,  to consolidate the two MockServer, into Test.Utilities project.
The MockServer under NuGet.CommandLine.Test needs an extra NuGet.Core package  https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGet.CommandLine.Test.csproj#L12-L15, while when I tried to add that package to Test.Utilities.csproj, there are many conflicts, e.g. ILogger is previously from NuGet.Common, but now it's from the newly added NuGet.Core package.
I haven't looked into why we need the old NuGet.Core package. But will do that in a separate PR if possible.

Pls let me know if it's okey. If yes, I'll create a tracking issue to refactor the code to consolidate the two MockServer.  Thanks!

Reply:
Thanks! Created tracking issue: https://github.com/NuGet/Client.Engineering/issues/949